pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/PersistentTopicsImpl.java

```diff
@@ -676,25 +680,57 @@ private Message getMessageFromHttpResponse(Response response) throws Exception {
 
             Map<String, String> properties = Maps.newTreeMap();
             MultivaluedMap<String, Object> headers = response.getHeaders();
-            Object publishTime = headers.getFirst("X-Pulsar-publish-time");
-            if (publishTime != null) {
-                properties.put("publish-time", (String) publishTime);
+            Object tmp = headers.getFirst("X-Pulsar-publish-time");
+            if (tmp != null) {
+                properties.put("publish-time", (String) tmp);
+            }
+            tmp =  headers.getFirst(BATCH_HEADER);
+            if (response.getHeaderString(BATCH_HEADER) != null) {
+                properties.put(BATCH_HEADER, (String)tmp);
+                return getIndividualMsgsFromBatch(msgId, data, properties);
```

Comment:
if we are returning here then does it mean we will not return `properties` for batch-messages? is it because we get properties while deserializing batch-msg?